lib/yggdrasil/commands.ex

```diff
@@ -0,0 +1,110 @@
+defprotocol Yggdrasil.Command do
+  def execute(player, cmd)
+end
+
+defmodule Yggdrasil.Command.MoveCommand do
+  defstruct exit: nil
+end
+
+defimpl Yggdrasil.Command, for: Yggdrasil.Command.MoveCommand do
```

Comment:
Awesome, love the use of protocols. Seems like the right place to use them. 

One thing though, I believe that you could actually define this inside the module like this

```elixir
defmodule Yggdrasil.Command.MoveCommand do
  defstruct exit: nil

  defimpl Yggdrasil.Command, for: Yggdrasil.Command.MoveCommand do
    def execute(player, cmd) do
      Yggdrasil.Player.notify player, "You head towards #{cmd.exit}"
    end
  end
end
```
Generally how I have seen it/done it.